offload/src/omptarget.cpp

```diff
@@ -461,7 +461,9 @@ void targetFreeExplicit(void *DevicePtr, int DeviceNum, int Kind,
   if (!DeviceOrErr)
     FATAL_MESSAGE(DeviceNum, "%s", toString(DeviceOrErr.takeError()).c_str());
 
-  DeviceOrErr->deleteData(DevicePtr, Kind);
+  if (DeviceOrErr->deleteData(DevicePtr, Kind))
```

Comment:
@jhuber6 Now if the dataDelete op fails, the program will be aborted. This causes tests like [offload/test/api/omp_host_pinned_memory.c](https://github.com/llvm/llvm-project/blob/main/offload/test/api/omp_host_pinned_memory.c) to fail if the op fails.

Reply:
Check it against `OFFLOAD_FAIL` instead of zero.

Reply:
Done.